src/main/java/com/google/sps/servlets/SignUpPageServlet.java

```diff
@@ -0,0 +1,66 @@
+package com.google.sps.servlets;
+
+import com.google.cloud.datastore.Datastore;
+import com.google.cloud.datastore.DatastoreOptions;
+import com.google.cloud.datastore.Entity;
+import com.google.cloud.datastore.FullEntity;
+import com.google.cloud.datastore.KeyFactory;
+import com.google.cloud.datastore.Key;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.Cookie;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.sql.rowset.WebRowSet;
+
+
+import org.jsoup.Jsoup;
+import org.jsoup.safety.Whitelist;
+
+@WebServlet("/signup")
+public class SignUpPageServlet extends HttpServlet {
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.sendRedirect("/signUp.html");
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    
+    String username = Jsoup.clean(request.getParameter("username"), Whitelist.none());
+    String email = Jsoup.clean(request.getParameter("email"), Whitelist.none());
+    String password = Jsoup.clean(request.getParameter("password"),Whitelist.none());
+
+    if(username!="" && email!="" && password!=""){
```

Comment:
Can you do the inverse of this condition?

if (username=="" .....) {return;}

Datastore datastore ....
.....